tests/integration_tests/conftest.py

```diff
@@ -84,39 +88,45 @@ def session_cloud():
     cloud.destroy()
 
 
-@pytest.fixture(scope='session', autouse=True)
-def setup_image(session_cloud):
+def get_validated_source(
+    source=integration_settings.CLOUD_INIT_SOURCE
+) -> CloudInitSource:
+    if source == 'NONE':
+        return CloudInitSource.NONE
+    elif source == 'IN_PLACE':
+        if session_cloud.datasource not in ['lxd_container', 'lxd_vm']:
+            raise ValueError(
+                'IN_PLACE as CLOUD_INIT_SOURCE only works for LXD')
+        return CloudInitSource.IN_PLACE
+    elif source == 'PROPOSED':
+        return CloudInitSource.PROPOSED
+    elif source.startswith('ppa:'):
+        return CloudInitSource.PPA
+    elif os.path.isfile(str(source)):
+        return CloudInitSource.DEB_PACKAGE
+    raise ValueError(
+        'Invalid value for CLOUD_INIT_SOURCE setting: {}'.format(source))
+
+
+@pytest.fixture(scope='session')
+def setup_image(session_cloud: IntegrationCloud):
     """Setup the target environment with the correct version of cloud-init.
 
     So we can launch instances / run tests with the correct image
     """
-    client = None
+    source = get_validated_source()
+    if source not in [
+        CloudInitSource.PROPOSED,
+        CloudInitSource.PPA,
+        CloudInitSource.DEB_PACKAGE,
+    ]:
+        return
```

Comment:
A nit: Should we invert this check? I think `NONE` and `IN_PLACE` are the exceptions to the rule: future `CloudInitSource`s are more likely to need setup than not (e.g. `CloudInitSource.RPM_PACKAGE`, `CloudInitSource.WHATEVER_RPM_DISTROS_USE_INSTEAD_OF_PPAS`, etc.) and it'll save us a manual update of this list in future.